code/modules/mob/living/simple_animal/hostile/mining_mobs/elites/elite.dm

```diff
@@ -0,0 +1,305 @@
+	icon = 'icons/mob/lavaland/elite_lavaland_monsters.dmi'
+	faction = list("mining_elite")
+	robust_searching = TRUE
+	ranged_ignores_vision = TRUE
+	ranged = 1
+	obj_damage = 5
+	var/chosen_attack = 1
+	vision_range = 6
+	aggro_vision_range = 18
+	var/list/attack_action_types = list()
+	environment_smash = ENVIRONMENT_SMASH_NONE  //This is to prevent elites smashing up the mining station, we'll make sure we can smash minerals fine below.
+	harm_intent_damage = 0 //Punching elites gets you nowhere
+	var/obj/structure/elite_crevice/myparent = null
+	var/can_talk = 0
+	stat_attack = UNCONSCIOUS
+	sentience_type = SENTIENCE_BOSS
+	
+		
+//Gives player-controlled variants the ability to swap attacks
+/mob/living/simple_animal/hostile/asteroid/elite/Initialize(mapload)
+	. = ..()
+	for(var/action_type in attack_action_types)
+		var/datum/action/innate/elite_attack/attack_action = new action_type()
+		attack_action.Grant(src)
+
+//Prevents elites from attacking members of their faction (can't hurt themselves either) and lets them mine rock with an attack despite not being able to smash walls.
+/mob/living/simple_animal/hostile/asteroid/elite/AttackingTarget()
+	if(istype(target, /mob/living/simple_animal/hostile))
+		var/mob/living/simple_animal/hostile/M = target
+		if(M.faction == src.faction)
+			return FALSE
+	. = ..()
+	if(ismineralturf(target))
+		var/turf/closed/mineral/M = target
+		M.gets_drilled()
+		
+//Elites can't talk (normally)!
+/mob/living/simple_animal/hostile/asteroid/elite/say(message, bubble_type, var/list/spans = list(), sanitize = TRUE, datum/language/language = null, ignore_spam = FALSE, forced = null)
+	if(can_talk == 1)
+		. = ..()
+		return
+	return 0
+		
+/*Basic setup for elite attacks, based on Whoneedspace's megafauna attack setup.
+While using this makes the system rely on OnFire, it still gives options for timers not tied to OnFire, and it makes using attacks consistent accross the board for player-controlled elites.*/
+
+/datum/action/innate/elite_attack
+	name = "Elite Attack"
+	icon_icon = 'icons/mob/actions/actions_elite.dmi'
+	button_icon_state = ""
+	var/mob/living/simple_animal/hostile/asteroid/elite/M
+	var/chosen_message
+	var/chosen_attack_num = 0
+
+/datum/action/innate/elite_attack/Grant(mob/living/L)
+	if(istype(L, /mob/living/simple_animal/hostile/asteroid/elite))
+		M = L
+		return ..()
+	return FALSE
+
+/datum/action/innate/elite_attack/Activate()
+	M.chosen_attack = chosen_attack_num
+	to_chat(M, chosen_message)
+	
+/mob/living/simple_animal/hostile/asteroid/elite/Life()
+	. = ..()
+	if(isturf(loc))
+		for(var/obj/structure/elite_crevice/crevice in loc)
+			if(crevice == myparent && myparent.activity == 2)
+				adjustHealth(-maxHealth*0.05)
+				var/obj/effect/temp_visual/heal/H = new /obj/effect/temp_visual/heal(get_turf(src))
+				H.color = "#FF0000"
+	if(myparent)
+		if(myparent.activity == 1 && myparent.activator.stat == DEAD)
```

Comment:
activity is, I'm assuming a bool. do (myparent.activity)

Reply:
nvm its not a bool. Use defines to make flags. This is confusing.

Reply:
Bools added in the upcoming fix, which should hopefully make this much easier to read.